### PR TITLE
[CI] Use libomp 13.0.0 to test XGBoost on MacOS

### LINKED ISSUE #7039: Segmentation fault on MacOS with libomp 12.0.0
The following code results in a Python segmentation fault when executed with Python 3.x on MacOS with libomp 12.0.0. Downgrading to a previous version of libomp solves it.

```
# xgboost macos seg fault when libomp==12.0.0
# passes with libomp==11.1.0

"""
	verify libomp version
		brew list --version libomp
		libomp 12.0.0
	to install libomp 11.1.0
		wget https://raw.githubusercontent.com/chenrui333/homebrew-core/0094d1513ce9e2e85e07443b8b5930ad298aad91/Formula/libomp.rb
        brew unlink libomp
        brew install --build-from-source ./libomp.rb
        brew list --version libomp

"""

import logging
import numpy as np
import xgboost as xgb

def xgboost_unit_tests():
	feature_arr = np.random.uniform(size=(100,5))
	label_arr = np.random.randint(low=0, high=10, size=(100,1))
	data_arr = np.concatenate((feature_arr, label_arr), axis=1)
	#Run Attribute Rank with -O 2
	run_xgboost(data_arr,data_arr)
	logging.info("XGBoost complete")
	return


def run_xgboost(trainarr, valarr):
	trainfeats = trainarr[:,:-1]
	trainlabels = trainarr[:,-1]
	# first seg fault
	dtrain = xgb.DMatrix(trainfeats, label=trainlabels)
	valfeats = valarr[:,:-1]
	vallabels = valarr[:,-1]
	# second seg fault
	dval = xgb.DMatrix(valfeats, label=vallabels)
	return


# Press the green button in the gutter to run the script.
if __name__ == '__main__':
	xgboost_unit_tests()
```

### PR DESCRIPTION
**Pull request by hcho3**
Closes #7039 

TODOs
- [ ] In the installation doc, add a warning: "If you are using Conda in MacOS, do not use `pip install xgboost`; this may lead to random crashes. Install XGBoost using Conda for best results."

### CODE REVIEW

File: .github/workflows/python_tests.yml [hunk 1]
```diff
@@ -17,10 +17,7 @@ jobs:
     - name: Install osx system dependencies
       if: matrix.os == 'macos-10.15'
       run: |
-        # Use libomp 11.1.0: https://github.com/dmlc/xgboost/issues/7039
-        wget https://raw.githubusercontent.com/Homebrew/homebrew-core/679923b4eb48a8dc7ecc1f05d06063cd79b3fc00/Formula/libomp.rb -O $(find $(brew --repository) -name libomp.rb)
         brew install ninja libomp
```
Comment (trivialfis) [inline .github/workflows/python_tests.yml:20]: If we are using conda, do we still need brew install libomp?

Comment (hcho3) [reply .github/workflows/python_tests.yml:20]: I don't think we can assume everything is using Conda.

**Review by trivialfis — APPROVED** [review]

**Comment by hcho3** [comment]
Let me update the docs too.

**Comment by hcho3** [comment]
I want to avoid https://github.com/dmlc/xgboost/issues/7039#issuecomment-1025038228

**Comment by hcho3** [comment]
Closing in favor of #7621

**Pull request closed without merging**